tests/functional/test_cli.py

```diff
@@ -0,0 +1,227 @@
+scenarios effectively.
+
+The purpose of this module is to verify that the CytoSnake CLI operates as expected and
+provides accurate results when interacting with various user parameters and modes. It
+validates that the CLI can handle different inputs, execute successfully in positive
+cases, and properly report errors in negative cases.
+
+The test scenarios are categorized into two main types:
+
+Positive Cases:
+These test cases validate the expected behavior of the CLI when
+provided with valid user parameters. The primary goal is to ensure that the CLI executes
+successfully without errors and produces the correct output.
+
+Negative Cases:
+These test cases simulate scenarios where invalid user parameters are provided.
+The objective is to confirm that the CLI can identify and handle errors appropriately,
+providing informative error messages to the user.
+"""
+
+import os
+import pathlib
+import shutil
+import subprocess
+
+import pytest
+
+from cytosnake.common import errors
+from cytosnake.utils import test_utils
+
+
+# ---------------
+# PyTest Fixtures
+# ---------------
+@pytest.fixture
+def testing_dir(tmp_path, request):
+    """Creates a testing directory
+
+    Note: Pytest will tear down tmp_path per test
+
+    Parameters
+    ----------
+    tmp_path : pytest.fixture
+        pytest default fixture value to be called when creating a temp dir.
+
+    request : pytest.fixture
+        Allows custom testing function to be added in the testing workflow
+
+    returns:
+    --------
+    pathlib.Path
+        Path pointing to temporary directory
+    """
+    # setting paths
+    original_cwd = str(pathlib.Path(".").absolute())
+
+    # creating a temporary path
+    test_name = request.node.name
+    tmp_dir = (tmp_path / test_name).absolute()
+    tmp_dir.mkdir()
+
+    # return the temporary directory path
+    yield tmp_dir
+
+    # teardown: remove the testing
+    if os.path.exists(tmp_dir):
+        shutil.rmtree(tmp_dir)
+
+    #  change the current working directory to the oroginal root
+    os.chdir(str(original_cwd))
+
+
+# ---------------
+# Init Mode Tests
+# ---------------
+@pytest.mark.negative
+def test_multiplate_maps_no_barcode(testing_dir) -> None:
+    """Negative case test: Expects `BarcodeMissingError` and Non-zero return code
+
+    This tests checks if the CLI raises an error when a user provides multiple platemaps
+    but no barcodes.
+
+    Rational:
+    ---------
+        CytoSnake relies on a barcode file to accurately map each platemap to its
+        corresponding plate. If users provide multiple platemaps without a barcode,
+        it becomes impossible for CytoSnake to determine the correct association between
+        platemaps and their respective plates.
+
+    Parameters:
+    -----------
+    testing_dir: pytest.fixture
+        Testing directory
+    """
+
+    # transfer data to testing folder
+    test_utils.prepare_dataset(test_data_name="nf1-data", test_dir_path=testing_dir)
```

Comment:
Seeing how this pattern repeats, would this be a good opportunity for a fixture?

Reply:
I was thinking about this earlier and I feel like the design of using fixture that require parameters are a bit daunting and may be not easy for new developers. 

Below is an example if `prepare_dataset()` becomes a fixture:
```python
@pytest.mark.parametrize("prepare_data", ["nf1"], indirect=True)
def text_example(testing_dir, prepare_dataset) -> None:
    # testing code below
```

However, if a redesign of the testing framework is conducted, then I will consider this approach.